### ISSUE TITLE
I want to change the loss during multi label classification.

### ISSUE THREAD

**Issue by Yongtae723** [huggingface/setfit#267]
Hi!

As you mention in readme, setfit will automatically use CrossEntropyLoss  and softmax with argmax when num_classes is greater than 1, if I use a differentiable head.

However, this setting is not suitable for multilabel classification.
Typically, sigmoid function and BCE loss is used.

So I want to modify some like below

if `self.model.multi_target_strategy ` is not None, setfit automatically

- return BCE loss on `model_head.get_loss_fn()`
- use the sigmoid function on forward of `SetFitHead` 

Can I do and make PR for you?

**Comment by tomaarsen** [comment]
#211 and #212 also tackle this issue. However, they have been proposed just prior to some fairly notable changes to SetFit, so it is plagued with merge conflicts right now. If you want, you can copy that branch and try to merge main into it. But feel free to start from scratch, too. I would however at least take the tests from that PR.

Perhaps you can go for `BCEWithLogitsLoss`? It should include a sigmoid already, and is recommended over `BCELoss` + sigmoid. Some notable issues that you may face is that the data is that the labels are `long`, while `BCEWithLogitsLoss` may require them to be floats, instead.

Feel free to work on this!

- Tom Aarsen

**Comment by Yongtae723** [comment]
I decided I will handle it!

But I have a question.
when I use copy the code of the #212, I would like to record them as their contributors. Can you change it so that?

**Comment by tomaarsen** [comment]
Wonderful!
I'm afraid I can't do that, but you can. There are a handful of approaches. I recommend the first one, as it will be more intuitive and easy, but there's also approach 2 if you prefer to work directly from #212 onwards.

## Approach 1: Manually copy/cherry-pick code from #212.
1. Prepare a new branch:
    ```
    git checkout main
    git pull upstream main
    git checkout -B feature/setfithead-multi-target
    ```
2. Manually copy/cherry-pick code from #212.
3. Add as co-author:
    Whenever you commit with copied code, add OskarLiew as a co-author using [these steps](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors). In short, it involves adding `Co-authored-by: OskarLiew <oskar@liew.se>` at the very end of the commit message.

For an example, I used this approach in this commit to add someone as a co-author: https://github.com/huggingface/setfit/pull/231/commits/2a7864feccfa711d858a8c8fc2a9014e4a33680b

## Approach 2: Continue directly from #212.
1. Fetch and checkout the branch of PR #212:
    ```
    git fetch upstream pull/212/head:feature/setfithead-multi-target
    git checkout feature/setfit-multi-target
    ```
    The first command fetches the PR with ID 212 and places it in a branch named `feature/setfithead-multi-target`, assuming that upstream points to the `huggingface/setfit` repository. See [this](https://stackoverflow.com/a/30584951/17936326) SO answer for info.
5. Pull upstream main into the new branch.
    ```
    git pull upstream main
    ```
    This will show you several merge conflicts, which you would have to solve.
6. Make sure it all works.
7. Push to Yongtae723/setfit.
    ```
    git push
    ```
    This command will likely already try to push it to origin, which is probably the `Yongtae723/setfit` repository. It'll warn you about no such branch existing on origin, and then it'll show you a command to resolve that.

Note that this approach means that you have to fix a lot of merge conflicts, and it won't be particularly easy.

Hope that helps.

- Tom Aarsen

**Comment by Yongtae723** [comment]
thank you for wanderful answering!

thanks to your explanation, I can understand it well!

thank you!
